app/products/calls/components/audio_device_button.tsx

```diff
@@ -0,0 +1,141 @@
+    currentCall: CurrentCall;
+}
+
+export const AudioDeviceButton = ({style, isLandscape, currentCall}: Props) => {
+    const intl = useIntl();
+    const theme = useTheme();
+    const {bottom} = useSafeAreaInsets();
+    const bold = typography('Body', 200, 'SemiBold');
+    const color = theme.awayIndicator;
+    const audioDeviceInfo = currentCall.audioDeviceInfo;
+    const earpieceLabel = intl.formatMessage({id: 'mobile.calls_earpiece', defaultMessage: 'Earpiece'});
+    const speakerLabel = intl.formatMessage({id: 'mobile.calls_speaker', defaultMessage: 'SpeakerPhone'});
+    const bluetoothLabel = intl.formatMessage({id: 'mobile.calls_bluetooth', defaultMessage: 'Bluetooth'});
+
+    const deviceSelector = useCallback(async () => {
+        const currentDevice = audioDeviceInfo.selectedAudioDevice;
+        const available = audioDeviceInfo.availableAudioDeviceList;
+        const selectDevice = (device: AudioDevice) => {
+            setPreferredAudioRoute(device);
+            dismissBottomSheet();
+        };
+
+        const renderContent = () => {
+            return (
+                <View>
+                    {available.includes(AudioDevice.Earpiece) &&
+                        <SlideUpPanelItem
+                            icon={'phone-in-talk'}
+                            onPress={() => selectDevice(AudioDevice.Earpiece)}
+                            text={earpieceLabel}
+                            textStyles={currentDevice === AudioDevice.Earpiece ? {...bold, color} : {}}
+                        />
+                    }
+                    {available.includes(AudioDevice.Speakerphone) &&
+                        <SlideUpPanelItem
+                            icon={'volume-high'}
+                            onPress={() => selectDevice(AudioDevice.Speakerphone)}
+                            text={speakerLabel}
+                            textStyles={currentDevice === AudioDevice.Speakerphone ? {...bold, color} : {}}
+                        />
+                    }
+                    {available.includes(AudioDevice.Bluetooth) &&
+                        <SlideUpPanelItem
+                            icon={'cellphone'}
+                            onPress={() => selectDevice(AudioDevice.Bluetooth)}
+                            text={bluetoothLabel}
+                            textStyles={currentDevice === AudioDevice.Bluetooth ? {...bold, color} : {}}
+                        />
+                    }
+                </View>
+            );
+        };
+
+        await bottomSheet({
+            closeButtonId: 'close-other-actions',
+            renderContent,
+            snapPoints: [1, bottomSheetSnapPoint(audioDeviceInfo.availableAudioDeviceList.length + 1, ITEM_HEIGHT, bottom)],
+            title: intl.formatMessage({id: 'mobile.calls_output_options', defaultMessage: 'Choose preferred output'}),
+            theme,
+        });
+    }, [setPreferredAudioRoute, audioDeviceInfo, bold, color]);
```

Comment:
```suggestion
    }, [setPreferredAudioRoute, audioDeviceInfo, color]);
```

as bold does not change and should be move outside of the component.